src/test/java/com/emc/object/s3/S3JerseyClientTest.java

```diff
@@ -199,6 +210,228 @@ public void testCreateEncryptedBucket() {
+            client.deleteVersion(bucketName, key, versionId);
+        }
+    }
+
+    @Test
+    public void testPutObjectLegalHold() throws Exception {
+        Assume.assumeTrue("Skip Object Lock related tests for non IAM user.", testIAM);
+
+        String bucketName = getTestBucket();
+        String key = "testObject_PutObjectLegalHold";
+        client.enableObjectLock(bucketName);
+
+        //Put Legal Hold on create
+        ObjectLockLegalHold objectLockLegalHold = new ObjectLockLegalHold().withStatus(ObjectLockLegalHold.Status.ON);
+        PutObjectRequest putObjectRequest = new PutObjectRequest(bucketName, key,"test Put Object LegalHold")
+                .withObjectLockLegalHold(objectLockLegalHold);
+        client.putObject(putObjectRequest);
+        String versionId = client.listVersions(bucketName, key).getVersions().get(0).getVersionId();
+        GetObjectLegalHoldRequest getObjectLegalHoldRequest = new GetObjectLegalHoldRequest(bucketName, key).withVersionId(versionId);
+        Assert.assertEquals("ON", client.getObjectLegalHold(getObjectLegalHoldRequest).getStatus());
+
+        //Put Legal Hold on existing object
+        objectLockLegalHold.setStatus(ObjectLockLegalHold.Status.OFF);
+        SetObjectLegalHoldRequest request = new SetObjectLegalHoldRequest(bucketName, key).withVersionId(versionId).withLegalHold(objectLockLegalHold);
+        client.setObjectLegalHold(request);
+        Assert.assertEquals("OFF", client.getObjectLegalHold(getObjectLegalHoldRequest).getStatus());
+    }
+
+    @Test
+    public void testPutObjectRetention() throws Exception {
+        Assume.assumeTrue("Skip Object Lock related tests for non IAM user.", testIAM);
+
+        String bucketName = getTestBucket();
+        String key = "testObject_PutObjectRetention";
+        client.enableObjectLock(bucketName);
+        Date retentionDate = new Date(System.currentTimeMillis() + 2000);
+        ObjectLockRetention objectLockRetention = new ObjectLockRetention()
+                .withMode(ObjectLockRetentionMode.COMPLIANCE)
+                .withRetainUntilDate(retentionDate);
+        //Put Retention on Create
+        PutObjectRequest putObjectRequest = new PutObjectRequest(bucketName, key,"test Put Object Retention")
+                .withObjectLockRetention(objectLockRetention);
+        client.putObject(putObjectRequest);
+        String versionId = client.listVersions(bucketName, key).getVersions().get(0).getVersionId();
+        GetObjectRetentionRequest request = new GetObjectRetentionRequest(bucketName,key).withVersionId(versionId);
+        ObjectLockRetention objectLockRetention2 = client.getObjectRetention(request);
+        Assert.assertEquals(objectLockRetention.getMode(), objectLockRetention2.getMode());
+        Assert.assertEquals(0, retentionDate.compareTo(objectLockRetention2.getRetainUntilDate()));
```

Comment:
you can `assertEquals()` the `Date`s directly

Reply:
Fixed.